### PR TITLE
salmon: patching to build with %gcc@13:

### PR DESCRIPTION
**Pull request by A-N-Other**
Adding a patch file to allow `salmon` to build with `gcc@13:`.

### CODE REVIEW

**Comment by A-N-Other** [comment]
This fix is now merged upstream, so switching patch to pull from GitHub commit. https://github.com/COMBINE-lab/salmon/pull/856

File: var/spack/repos/builtin/packages/salmon/package.py [hunk 1]
```diff
@@ -85,6 +85,13 @@ class Salmon(CMakePackage):
             when="@{0}".format(ver),
         )
 
+    patch(
+        "https://github.com/COMBINE-lab/salmon/commit/ffb2a11.patch?full_index=1",
+        sha256="5ed3512bae665c1d72002911ab9ee6d213f10df63019ebd9e8e0ecde03823a73",
+        when="%gcc@13:",
```
Comment (snehring) [inline var/spack/repos/builtin/packages/salmon/package.py]: If they've got it merged in such that it's likely it'll be in a future release, could you add a version restriction to the patch?

Comment (A-N-Other) [reply var/spack/repos/builtin/packages/salmon/package.py]: Done!

**Review by snehring — APPROVED** [review]
lgtm, builds with gcc 13.1.0 on rhel9.

File: var/spack/repos/builtin/packages/salmon/package.py [hunk 2]
```diff
@@ -85,6 +85,13 @@ class Salmon(CMakePackage):
             when="@{0}".format(ver),
         )
 
+    patch(
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/salmon/package.py:90]: Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed

?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/salmon/package.py (hunk 2 above)
Comment (A-N-Other) [reply var/spack/repos/builtin/packages/salmon/package.py:90]: Done 👍

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@spackbot fix style

**Comment by alalazo** [comment]
@A-N-Other Can you fix style? I didn't notice you are submitting PR from an organization, when doing so GitHub doesn't allow to give maintainers permissions to push etc. to your branch

**Comment by A-N-Other** [comment]
@spackbot fix style

Line lengths 👀

**Comment by alalazo** [comment]
I think you need to do it manually :sweat: There's a trailing whitespace.

**Comment by A-N-Other** [comment]
@alalazo

Done. Don't know why spackbot didn't like it - normally works for me. Must have done something different when I submitted it ?!?

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Are you always submitting from https://github.com/LMSBioinformatics ? If I read [this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork) correctly permissions can be given only on user-owned forks (so, I guess, not from organizations)

**Comment by A-N-Other** [comment]
Yeah, I have a spack/spack clone there that I play with, so they're always coming from the same place. No matter.

**Pull request merged**